sentry/src/test/java/io/sentry/SentryTracerTest.kt

```diff
@@ -327,4 +331,33 @@ class SentryTracerTest {
 
         assertNotNull(transaction.startTimestamp)
     }
+
+    @Test
+    fun `when waiting for children, finishing transaction does not call hub if all children are not finished`() {
+        val transaction = fixture.getSut(waitForChildren = true)
+        transaction.startChild("op")
+        transaction.finish()
+        verify(fixture.hub, never()).captureTransaction(any())
+    }
+
+    @Test
+    fun `when waiting for children, finishing transaction calls hub if all children are finished`() {
+        val transaction = fixture.getSut(waitForChildren = true)
+        val child = transaction.startChild("op")
+        child.finish()
+        transaction.finish()
```

Comment:
if you remove this line, the test is still green I guess, but `transaction.finish()` should be called too

Reply:
Thanks, same has been caught by @brustolin. Test added and code fixed.